mmdeploy/codebase/mmdet3d/deploy/voxel_detection.py

```diff
@@ -50,107 +76,142 @@ def build_pytorch_model(self,
+
+        model = build_voxel_detection_model(
+            model_files,
+            self.model_cfg,
+            self.deploy_cfg,
+            device=self.device,
+            data_preprocessor=data_preprocessor)
+        model = model.to(self.device)
+        return model
 
-    def create_input(self, pcd: str, *args) -> Tuple[Dict, torch.Tensor]:
+    def create_input(
+        self,
+        pcd: str,
+        input_shape: Sequence[int] = None,
+        data_preprocessor: Optional[BaseDataPreprocessor] = None
+    ) -> Tuple[Dict, torch.Tensor]:
         """Create input for detector.
 
         Args:
             pcd (str): Input pcd file path.
+            input_shape (Sequence[int], optional): model input shape.Defaults to None. # noqa: E501
+            data_preprocessor (Optional[BaseDataPreprocessor], optional): model input preprocess. Defaults to None. # noqa: E501
 
         Returns:
             tuple: (data, input), meta information for the input pcd
                 and model input.
         """
-        data = VoxelDetection.read_pcd_file(pcd, self.model_cfg, self.device)
-        voxels, num_points, coors = VoxelDetectionModel.voxelize(
-            self.model_cfg, data['points'][0])
-        return data, (voxels, num_points, coors)
+
+        preproc = self.build_data_preprocessor()
+
+        cfg = self.model_cfg
+        test_pipeline = deepcopy(cfg.test_dataloader.dataset.pipeline)
+        test_pipeline = Compose(test_pipeline)
+        box_type_3d, box_mode_3d = \
+            get_box_type(cfg.test_dataloader.dataset.box_type_3d)
+
+        data = []
+        data_ = dict(
+            lidar_points=dict(lidar_path=pcd),
+            timestamp=1,
+            # for ScanNet demo we need axis_align_matrix
+            axis_align_matrix=np.eye(4),
+            box_type_3d=box_type_3d,
+            box_mode_3d=box_mode_3d)
+        data_ = test_pipeline(data_)
+        data.append(data_)
+
+        collate_data = pseudo_collate(data)
+        data[0]['inputs']['points'] = data[0]['inputs']['points'].to(
+            self.device)
+
+        if preproc is not None:
+            collate_data = preproc(collate_data, False)
+            del collate_data['inputs']['voxels']['voxel_centers']
+
+        voxels = collate_data['inputs']['voxels']
+        inputs = [voxels['voxels'], voxels['num_points'], voxels['coors']]
+        return collate_data, inputs
 
     def visualize(self,
+                  image: Union[str, np.ndarray],
                   model: torch.nn.Module,
-                  image: str,
                   result: list,
                   output_file: str,
-                  window_name: str,
+                  window_name: str = '',
                   show_result: bool = False,
-                  score_thr: float = 0.3):
-        """Visualize predictions of a model.
+                  draw_gt: bool = False,
+                  data_samples=None,
```

Comment:
could remove this `data_samples` argument since it is unused.

Reply:
done.